serve/mlc_serve/engine/staging_engine_worker.py

```diff
@@ -293,15 +296,21 @@ def _adjust_batch(self):
                 state = self.queue[0]
                 num_tokens = len(state.token_ids)
                 num_new_batched_tokens += num_tokens
+                # this can happen if we processed some request and then had to remove it from inference
+                # due to limit of the kv cache. It can appear that new size to process is bigger
+                # than max_num_batched_tokens. Need to roll back to acceptable size
```

Comment:
Here is my proposal for the revised text based on my understanding:

"""
This can happen when we are recovering from cache eviction and the sum of prompt and intermediate decode tokens is bigger than the biggest allowable batch size, `self.max_num_batched_tokens`. In such cases, we need to discard the recent decode tokens that cannot fit into a batch, and recompute them after we fill the cache entries for the older tokens. 
"""